src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -94,6 +94,81 @@
     }
     
     /**
+     * Generates the email containing the summary of the feedback sessions
+     * email for the given {@code courseId} for {@code student}
+     * @param courseId - ID of the course
+     * @param student - attributes of student to send feedback session summary to
+     */
+    public EmailWrapper generateFeedbackSessionSummaryOfCourse(String courseId, StudentAttributes student) {
+        
+        CourseAttributes course = coursesLogic.getCourse(courseId);
+        
+        List<FeedbackSessionAttributes> sessions = new ArrayList<FeedbackSessionAttributes>();
+        List<FeedbackSessionAttributes> fsInCourse = fsLogic.getFeedbackSessionsForCourse(courseId);
+        
+        for (FeedbackSessionAttributes fsa : fsInCourse) {
+            if (!fsa.isPrivateSession() && (fsa.isSentOpenEmail() || fsa.isSentPublishedEmail())) {
+                sessions.add(fsa);
+            }
+        }
+        
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String joinUrl = Config.getAppUrl(student.getRegistrationUrl()).toAbsoluteString();
+        
+        String joinFragmentValue = isYetToJoinCourse(student)
+                                   ? Templates.populateTemplate(EmailTemplates.FRAGMENT_STUDENT_COURSE_JOIN,
+                                           "${joinUrl}", joinUrl,
+                                           "${courseName}", course.getName())
+                                   : "";
+        
+        for (FeedbackSessionAttributes fsa : sessions) {
+            
+            String submitUrlHtml = "(Feedback session is " + (fsa.isClosed() ? "closed" : "not yet opened") + ")";
+            String reportUrlHtml = "(Feedback session is not yet published)";
+            
+            if (fsa.isOpened()) {
+                String submitUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                        .withCourseId(course.getId())
+                        .withSessionName(fsa.getFeedbackSessionName())
+                        .withRegistrationKey(StringHelper.encrypt(student.key))
+                        .withStudentEmail(student.email)
+                        .toAbsoluteString();
+                submitUrlHtml = "<a href=\"" + submitUrl + "\">" + submitUrl + "</a>";
+                
+                String reportUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_RESULTS_PAGE)
```

Comment:
Looks like I missed one here.
An open session does not mean it is published. The correct check for this block should be `fsa.isPublished()`.